format-specs/geoparquet.md

```diff
@@ -53,7 +53,7 @@ Each geometry column in the dataset must be included in the columns field above
 | Field Name | Type | Description |
 | --- | --- | --- |
 | encoding | string | **REQUIRED** Name of the geometry encoding format. Currently only 'WKB' is supported. |
-| geometry_type | string or \[string] | **REQUIRED** The geometry type(s) of all geometries, or 'Unknown' if they are not known. |
+| geometry_types | \[string] | **REQUIRED** The geometry types of all geometries, or an empty array if they are not known. |
```

Comment:
If the field can be empty, why require it anyway?

Reply:
Also, the description should probably mention that it's a set of values (i.e. unique items in the array - see also the other comment)

Reply:
If it is optional, I assume we wouldn't assign any different meaning to `{}` and `{"geometry_types": []}`.  So then do we want `"minItems": 1`?

I like the idea of avoiding two ways to say the same thing.  No strong preference between optional with min length of one vs required.  Though it seems slightly more explicit to say that `[]` means "unknown" (whereas the absence of the property might either mean "unknown" or "I forgot" - not sure if there is any significant difference between the two though).

Reply:
I agree with the point I think you are making though.  If it isn't required that the geometry types be known, why require the geometry types in the metadata?

Reply:
Yeah, if it is optional, minItems should be set to 1, indeed. I have a slight tendency for not requiring it as for me an empty array reads more as "no type", but requiring it makes people think (if implementations don't set a default such as  `write_geoparquet(geometry_types = [], ...)`

Reply:
Some discussion on this topic happened in the original issue https://github.com/opengeospatial/geoparquet/issues/41. The idea was indeed that we wanted to avoid two ways to say the same thing (not specified if optional, vs `geometry_type="Unknown"`. 

And I think the rationale for making it required was indeed to have people think about it / recommend to specify a useful value (instead of just leaving out): https://github.com/opengeospatial/geoparquet/issues/41#issuecomment-1079157251

I also find the empty array a bit less clear in indicating "unknown". Another option could also be to have the field required and allow either the single string "Unknown" or either a list of minimum length 1.

Reply:
```go
len(geometry_types) == 0
```
vs
```go
geometry_types[0] = "Unknown"
```

(I misspelled Unknown twice while trying to type that :)

Reply:
Yeah, I'm not a big fan of "Unknown" either. Also, thinking a bit more about STAC it seems for everything we made required to make people think about it, it did not really work that way. People still just push things out without validation, so I could also see that people just omit geometry_types although required. So I tend towards not required and minLength: 1 for the array. As an explicit way to formulate "unknown" we could also consider "null".